functions/src/notifications/publishNotifications.ts

```diff
@@ -88,49 +93,20 @@ export const publishNotifications = functions.firestore
     console.log(`topic type: ${topic.type}`)
 
     const handleNotifications = async (
-      topic: BillNotification | OrgNotification
+      topic: BillHistoryUpdateNotification | TestimonySubmissionNotification
     ) => {
       const notificationFields = createNotificationFields(topic)
 
       console.log(JSON.stringify(notificationFields))
 
       const topicNameSnapshot = await db
         .collectionGroup("activeTopicSubscriptions")
-        .where("topicName", "==", notificationFields.topicName)
+        .where("topicName", "in", notificationFields.topicName)
```

Comment:
Are we sure this has solved the duplication problem? It looks like if we match on two rules for the same user, it would return two snapshots here (e.g. one for the bill topic and one for the org/user topic) and we would create one notification for each snapshot.

It might be helpful to:
* Wait to `createNotificationFields` until we know which topics we matched on 
* Group the returned `activeTopicSubscription`s by `uid`
* Extract the matching follow types (billFollow vs userFollow) from the `activeTopicSubscription`s 
* For each `uid`, run `createNotificationFields` only now that we know which follows types triggered the notification and can set `isBillMatch`/`isUserMatch` for each user's notificatiion accordingly.

Reply:
You are correct, it didn't solve the duplication problem. I thought it solved the duplication problem because the notifications didn't appear, but it's because the topicName that was being created was prefixed with org instead of testimony, so it never populated the notification. I've updated the code to reflect your suggestions.